README.md
Installing Behat-Rest
---------------------

The easiest way to install Behat is by using [Composer](https://getcomposer.org):

<pre>
$> curl -sS https://getcomposer.org/installer | php
$> php composer.phar require --dev hola/behat-rest
</pre>

Configuration Behat-Rest
------------------------

Create directory called "features" in root directory, into features directory create file "behat.yml" contains configuration: 

<pre>
default:
  suites:
    default:
      contexts:
        - Hola\Behat\WebApiContext:
          - https://url.api.to.test/uri/
      paths:
        - %paths.base%
</pre>
Always end url with /

News Features
-------------

Create new features into features folder and execute with behat:

<pre>
vendor/bin/behat -c features/behat.yml features/
</pre>

Sample Feature
--------------

Feature: Testing sample REST services
    In order to maintain user information through the services
    as a service user
    I want to see if the services work as expected 

    Scenario: Creating a New Employee
        When I send a POST request to "/employee" with values:
            | employeeId | 007         |
            | name       | James Bond  |
            | age        | 27          |
        Then response code should be 200
            And the response should be "true"

    Scenario: Finding an Existing Employee
        When I send a GET request to "/employee/007"
        Then response code should be 200
            And the response should contain json:
            """"
            {
                "name": "James Bond",
                "age": "27"
            }
            """"
        And in the response there is no field called "gender"

    Scenario: Updating an Existing Employee
        When I send a PUT request to "/employee/007" with values:
            | age | 38 |
        Then response code should be 200
            And the response should be "true"

    Scenario: Deleting Existing and Non-existing Employees
        Given I send a DELETE request to "/employee/007"
        Then response code should be 200
            And the response should be "true"
        Given I send a DELETE request to "/employee/008"
        Then response code should be 400
            And the response should contain "Unable to delete because the employee does not exist."



CHANGELOG.md
Behat_Rest-0.1.0 (04/08/2015)
==============================
- Initial Release

Behat_Rest-0.1.1 (07/08/2015)
==============================
- Renaming the context, mode change requests and make more features
- Control exceptions to http codes 400
- Resolve bugs to get body response

Behat_Rest-0.1.2 (07/08/2015)
==============================
- Change uri, now in config file, features files only need resource to test. To avoid problems

Behat_Rest-0.1.3 (12/08/2015)
==============================
- Add new function to compare response property with integers

Behat_Rest-0.1.4 (19/08/2015)
==============================
- Updated dependencies (`hola/configuration` to `0.2.*`)

Behat_Rest-0.1.5 (04/11/2015)
==============================
- Update composer packages to common versions

Behat_Rest-0.1.6 (02/12/2015)
==============================
- Add patch action

Behat_Rest-0.2.0 (19/05/2016)
=============================
- Updated dependencies (`hola/configuration` to `0.4.0-BETA1`)

Behat_Rest-0.2.1 (14/06/2016)
===============================
- Updated dependencies (`hola/configuration` to `0.4.*`)

Behat_Rest-0.2.2 (13/03/2017)
=============================
- Add functions when a GET is needed before the action

src/test/bootstrap.php
<?php
/**
 * <strong>Name : bootstrap.php</strong><br>
 * <strong>Desc : Bootstrapping application</strong><br>
 *
 * PHP version 5.3
 *
 * @category  Hello-canada
 * @package   Hello-canada
 * @author    Desarrollo <desarrollo@hola-internet.com>
 * @copyright 2013 Hola.com
 * @license   Apache 2 License http://www.apache.org/licenses/LICENSE-2.0.html
 * @version   GIT: $Id$
 * @link      http://www.hola.com
 * @since     File available since Release 0.1
 */

date_default_timezone_set('UTC');

putenv('PHP_ENV=testing');

$php_env = getenv('PHP_ENV');

if (empty($php_env)) {
    $php_env = 'testing';
    putenv("PHP_ENV={$php_env}");
}

// Checking composer autoload.
// Previous to ConfigMediator as the latter is spread across the former.
$loader = include __DIR__ . '/../../vendor/autoload.php';

$loader->setUseIncludePath(true);

// Instantiating Config Mediator
$configuration = new Hola\Configuration\ConfigMediator(__DIR__ . "/../../");

// Setting our common variables.
$configuration->pushConfigurationFromFiles(
    array(__DIR__ . "/../main/resources/config/config.yml"),
    array("common")
);

// Setting our environment variables.
$configuration->pushConfigurationFromFiles(
    array(
        $configuration->getProperty(
            "CONFIGURATION_FILES_".
            strtoupper($php_env)
        )
    ),
    array($php_env)
);

set_include_path(
    '.' .
    PATH_SEPARATOR . $configuration->getProperty("VENDOR_PATH") .
    PATH_SEPARATOR . $configuration->getProperty("TEST_PATH") .
    PATH_SEPARATOR . $configuration->getProperty("MOCKS_PATH") .
    PATH_SEPARATOR . get_include_path()
);

// Cleaning up ConfigMediator.
$configuration->popConfigurationsFromQueue(
    array_reverse($configuration->getFilesRead())
);

$_SERVER['REQUEST_METHOD'] = 'GET';


src/test/resources/mocks/MockProxy.php
<?php


class MockProxy {
  public static function setStaticExpectations($mock) {
    static::$mock = $mock;
  }
  
  public static function __callStatic($name, $args) {
    return call_user_func_array(
        array(static::$mock,$name),
        $args
    );
  }
}

src/main/Hola/Behat/WebApiContext.php
<?php
/**
 * <strong>Name :  WebApiContext</strong><br>
 * <strong>Desc :  Base Class to Web Api Context</strong><br>
 *
 * PHP version 5.5.20
 *
 * @category  BehatRest
 * @package   Hola\Behat
 * @author    Development <desarrollo@hola-internet.com>
 * @copyright 2015 Hola.com
 * @license   Apache 2 License http://www.apache.org/licenses/LICENSE-2.0.html
 * @version   GIT: $Id$
 * @link      http://www.hola.com
 * @since     File available since Release 0.1.0
 */

namespace Hola\Behat;

use Behat\Behat\Context\Context;
use GuzzleHttp\Client;
use GuzzleHttp\Psr7\Response;
use Behat\Gherkin\Node\TableNode;
use Behat\Gherkin\Node\PyStringNode;
use GuzzleHttp\Exception\RequestException;
use PHPUnit_Framework_Assert as Assertions;

/**
 * Class WebApiContext
 *
 * @category   Service
 * @subpackage WebApiContext
 * @author     Development <desarrollo@hola-internet.com>
 * @copyright  2015 Hola.com
 * @license    Apache 2 License http://www.apache.org/licenses/LICENSE-2.0.html
 * @version    Release: <0.1.0>
 * @link       http://www.hola.com
 * @since      Class available since Release 0.1.0
 */
class WebApiContext implements Context
{
    /**
     * Http response
     *
     * @access protected
     * @var Response
     */
    protected $response;


    /**
     * Http Client
     *
     * @access protected
     * @var Client
     */
    protected $client;


    /**
     * Url Api to test
     *
     * @access protected
     * @var string
     */
    protected $baseUrl = array();

    /**
     * Placeholders
     *
     * @access protected
     * @var array
     */
    protected $placeHolders = array();


    /**
     * Send request by Guzzle
     *
     * @param string $method  method http
     * @param string $uri     uri service
     * @param array  $options options to request
     * @param array  $headers optional headers to request
     *
     * @access protected
     * @return void
     */
    protected function sendRequest($method, $uri, $options, $headers = array())
    {
        $request = null;
        // Headers need services
        if (empty($headers)) {
            $headers = array(
                'headers' => array('Accept'       => 'application/json',
                                   'content-type' => 'application/json'
                )
            );
        }
        $options = array_merge($options, $headers);

        try {
            switch ($method) {
                case 'GET':
                    $request = $this->client->get($uri, $options);
                    break;
                case 'POST':
                    $request = $this->client->post($uri, $options);
                    break;
                case 'PUT':
                    $request = $this->client->put($uri, $options);
                    break;
                case 'PATCH':
                    $request = $this->client->patch($uri, $options);
                    break;
                case 'DELETE':
                    $request = $this->client->delete($uri, $options);
                    break;
            }

            $this->response = $request;
            // Control exception for http codes 4xx
        } catch (RequestException $exception) {
            $this->response = $exception->getResponse();
        }
    }

    /**
     * Replaces placeholders in provided text.
     *
     * @param string $string
     *
     * @return string
     */
    protected function replacePlaceHolder($string)
    {
        foreach ($this->placeHolders as $key => $val) {
            $string = str_replace($key, $val, $string);
        }

        return $string;
    }


    /**
     * Initializes context.
     *
     * @param string $baseUrl base url of service
     */
    public function __construct($baseUrl)
    {
        $this->baseUrl = $baseUrl;
        $client = new Client(['base_uri' => $this->baseUrl]);
        $this->client = $client;
    }

    /**
     * Sends HTTP request to specific relative URL.
     *
     * @param string $method     request method
     * @param string $uri        relative url
     * @param string $identifier identifier
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)" need GET "([^"]+)"$/
     */
    public function iSendARequestWithPreGet($method, $uri, $identifier = null)
    {
        $parts = parse_url($uri);

        if (!empty($identifier) && array_key_exists('query', $parts)) {
            $this->sendRequest('GET', $uri, array());
            $body = json_decode((string)$this->response->getBody(), true);
            $identifier = $body[0][$identifier];
            $uri = $parts['path'] . "/$identifier";
        }

        $this->sendRequest($method, $uri, array());
    }

    /**
     * Sends HTTP request to specific relative URL.
     *
     * @param string $method     request method
     * @param string $uri        relative url
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)"$/
     */
    public function iSendARequest($method, $uri)
    {
        $this->sendRequest($method, $uri, array());
    }

    /**
     * Sends HTTP request to specific relative URL accepting text/html
     *
     * @param string $method request method
     * @param string $uri    relative url
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)" Accept HTML$/
     */
    public function iSendARequestHTML($method, $uri)
    {
        $this->sendRequest(
            $method, $uri, array(), array(
            'headers' => array('Accept'       => 'text/html',
                               'content-type' => 'text/html'
            ))
        );
    }

    /**
     * Sends HTTP request to specific URL with field values from Table.
     *
     * @param string    $method     request method
     * @param string    $uri        relative uri
     * @param string    $identifier identifier to retrieve
     * @param TableNode $post       table of post values
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)" need GET "([^"]*)" with values:$/
     */
    public function iSendARequestWithValuesWithPreGet($method, $uri, $identifier = null, TableNode $post)
    {
        $fields = array();

        $parts = parse_url($uri);

        if (!empty($identifier) && array_key_exists('query', $parts)) {
            $this->sendRequest('GET', $uri, array());
            $body = json_decode((string)$this->response->getBody(), true);
            $identifier = $body[0][$identifier];
            $uri = $parts['path'] . "/$identifier";
        }

        foreach ($post->getRowsHash() as $key => $val) {
            $fields[$key] = $this->replacePlaceHolder($val);
        }

        $bodyOption = json_encode($fields);

        $this->sendRequest($method, $uri, array('body' => $bodyOption));
    }

    /**
     * Sends HTTP request to specific URL with field values from Table.
     *
     * @param string    $method request method
     * @param string    $uri    relative uri
     * @param TableNode $post   table of post values
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)" with values:$/
     */
    public function iSendARequestWithValues($method, $uri, TableNode $post)
    {
        $fields = array();

        foreach ($post->getRowsHash() as $key => $val) {
            $fields[$key] = $this->replacePlaceHolder($val);
        }

        $bodyOption = json_encode($fields);

        $this->sendRequest($method, $uri, array('body' => $bodyOption));
    }

    /**
     * Sends HTTP request to specific URL with raw body from PyString.
     *
     * @param string       $method request method
     * @param string       $uri    relative uri
     * @param PyStringNode $string request body
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)" with body:$/
     */
    public function iSendARequestWithBody($method, $uri, PyStringNode $string)
    {
        $bodyOption = $this->replacePlaceHolder(trim($string));

        $this->sendRequest($method, $uri, array('body' => $bodyOption));

    }

    /**
     * Sends HTTP request to specific URL with form data from PyString.
     *
     * @param string       $method request method
     * @param string       $uri    relative uri
     * @param PyStringNode $body   request body
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)" with form data:$/
     */
    public function iSendARequestWithFormData($method, $uri, PyStringNode $body)
    {
        $body = $this->replacePlaceHolder(trim($body));

        $fields = array();
        parse_str(implode('&', explode("\n", $body)), $fields);

        $this->sendRequest($method, $uri, array('body' => $fields));
    }

    /**
     * Sends HTTP request to specific URL with query params from PyString.
     *
     * @param string       $method request method
     * @param string       $uri    relative uri
     * @param PyStringNode $body   request body
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)" with query params:$/
     */
    public function iSendARequestWithQueryParams($method, $uri, PyStringNode $body)
    {
        $body = $this->replacePlaceHolder(trim($body));
        $fields = implode('&', explode("\n", $body));
        $this->sendRequest($method, $uri . '?' . $fields, array());
    }

    /**
     * Sends HTTP request to specific URL with query params from PyString.
     *
     * @param string       $method request method
     * @param string       $uri    relative uri
     * @param PyStringNode $body   request body
     *
     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]+)" Accept HTML with query params:$/
     */
    public function iSendAHTMLRequestWithQueryParams($method, $uri, PyStringNode $body)
    {
        $body = $this->replacePlaceHolder(trim($body));
        $fields = implode('&', explode("\n", $body));
        $this->sendRequest(
            $method, $uri . '?' . $fields, array(), array(
            'headers' => array('Accept'       => 'text/html',
                               'content-type' => 'text/html'
            ))
        );
    }

    /**
     * Check that response is a JSON
     *
     * @throws \Exception
     *
     * @Then /^the response should be JSON$/
     */
    public function theResponseShouldBeJson()
    {
        $data = json_decode($this->response->getBody(true));

        if (empty($data)) {
            throw new \Exception("Response was not JSON\n" . $this->response);
        }
    }

    /**
     * Check that response is a HTML
     *
     * @throws \Exception
     *
     * @Then /^the response should be HTML$/
     */
    public function theResponseShouldBeHtml()
    {
        $content_type = $this->response->getHeader('Content-Type');
        if (!in_array('text/html', $content_type)) {
            throw new \Exception("Response was not HTML\n" . json_encode($content_type));
        }
    }

    /**
     * Checks that response has specific status code.
     *
     * @param string $httpStatus status code
     *
     * @throws \Exception
     *
     * @Then /^the response status code should be (\d+)$/
     */
    public function theResponseStatusCodeShouldBe($httpStatus)
    {
        if ((string)$this->response->getStatusCode() !== $httpStatus) {
            throw new \Exception(
                'HTTP code does not match ' . $httpStatus .
                ' (actual: ' . $this->response->getStatusCode() . ')'
            );
        }
    }

    /**
     * Check that response has a property
     *
     * @param string $propertyName name property
     *
     * @throws \Exception
     *
     * @Then /^the response has a "([^"]*)" property$/
     */
    public function theResponseHasAProperty($propertyName)
    {
        $body = json_decode((string)$this->response->getBody(), true);
        if (array_key_exists(0, $body)) {
            $data = $body[0];
        } else {
            $data = $body;
        }


        if (!empty($data)) {
            if (!isset($data[$propertyName])) {
                throw new \Exception("Property '" . $propertyName . "' is not set!\n");
            }
        } else {
            throw new \Exception("Response was not JSON\n" . $this->response->getBody(true));
        }
    }


    /**
     * Check the property is equal to passed string
     *
     * @param string $propertyName  name property
     * @param string $propertyValue property value
     *
     * @throws \Exception
     *
     * @Then /^the "([^"]*)" property equals number (\d+)$/
     */
    public function thePropertyEqualsNumber($propertyName, $propertyValue)
    {
        $data = json_decode($this->response->getBody(true));

        if (!empty($data)) {
            if (!isset($data->$propertyName)) {
                throw new \Exception("Property '" . $propertyName . "' is not set!\n");
            }
            if ($data->$propertyName !== (int)$propertyValue) {
                throw new \Exception(
                    'Property value mismatch! (given: ' . $propertyValue . ', match: ' . $data->$propertyName . ')'
                );
            }
        } else {
            throw new \Exception("Response was not JSON\n" . $this->response->getBody(true));
        }
    }

    /**
     * Check the property is equal to passed string
     *
     * @param string $propertyName  name property
     * @param string $propertyValue property value
     *
     * @throws \Exception
     *
     * @Then /^the "([^"]*)" property equals "([^"]*)"$/
     */
    public function thePropertyEquals($propertyName, $propertyValue)
    {
        $data = json_decode($this->response->getBody(true));

        if (!empty($data)) {
            if (!isset($data->$propertyName)) {
                throw new \Exception("Property '" . $propertyName . "' is not set!\n");
            }
            if ($data->$propertyName != $propertyValue) {
                throw new \Exception(
                    'Property value mismatch! (given: ' . $propertyValue . ', match: ' . $data->$propertyName . ')'
                );
            }
        } else {
            throw new \Exception("Response was not JSON\n" . $this->response->getBody(true));
        }
    }

    /**
     * Checks that response body contains JSON from PyString.
     *
     * Do not check that the response body /only/ contains the JSON from PyString,
     *
     * @param PyStringNode $jsonString
     *
     * @throws \RuntimeException
     *
     * @Then /^(?:the )?response should contain json:$/
     */
    public function theResponseShouldContainJson(PyStringNode $jsonString)
    {
        $etalon = json_decode($this->replacePlaceHolder($jsonString->getRaw()), true);
        $actual = json_decode((string)$this->response->getBody(), true);

        if (null === $etalon) {
            throw new \RuntimeException(
                "Can not convert etalon to json:\n" . $this->replacePlaceHolder($jsonString->getRaw())
            );
        }

        Assertions::assertGreaterThanOrEqual(count($etalon), count($actual));
        foreach ($etalon as $key => $needle) {
            Assertions::assertArrayHasKey($key, $actual);
            Assertions::assertEquals($etalon[$key], $actual[$key]);
        }
    }

    /**
     * Checks that response body contains X items
     *
     * @param integer $items items response
     *
     * @throws \Exception
     *
     * @Then /^the response contains (\d+) item(s)$/
     */
    public function theResponseContainsItems($items)
    {
        $data = json_decode((string)$this->response->getBody(), true);

        if (!empty($data)) {
            if (count($data) != $items) {
                throw new \Exception("No contains $items items");
            }
        } else {
            throw new \Exception("Response was not JSON\n" . $this->response->getBody(true));
        }
    }
}


src/main/Hola/Behat/FeatureContext.php
<?php
/**
 * <strong>Name :  FeatureContext</strong><br>
 * <strong>Desc :  Base Feature Context</strong><br>
 *
 * PHP version 5.5.20
 *
 * @category  BehatRest
 * @package   Hola\Behat
 * @author    Development <desarrollo@hola-internet.com>
 * @copyright 2015 Hola.com
 * @license   Apache 2 License http://www.apache.org/licenses/LICENSE-2.0.html
 * @version   GIT: $Id$
 * @link      http://www.hola.com
 * @since     File available since Release 0.1.0
 */

namespace Hola\Behat;

use Behat\Behat\Context\Context;
use GuzzleHttp\Client;
use GuzzleHttp\Psr7\Response;

/**
 * Class FeatureContext
 *
 * @category   Service
 * @subpackage FeatureContext
 * @author     Development <desarrollo@hola-internet.com>
 * @copyright  ${YEAR} Hola.com
 * @license    Apache 2 License http://www.apache.org/licenses/LICENSE-2.0.html
 * @version    Release: <0.1.0>
 * @link       http://www.hola.com
 * @since      Class available since Release 0.1.0
 */
class FeatureContext implements Context
{
    /**
     * Http response
     *
     * @access protected
     * @var Response
     */
    protected $response;


    /**
     * Http Client
     *
     * @access protected
     * @var Client
     */
    protected $client;


    /**
     * Url Api to test
     *
     * @access protected
     * @var string
     */
    protected $baseUrl = array();


    /**
     * Initializes context.
     * Every scenario gets it's own context object.
     *
     * @param array $parameters context parameters (set them up through behat.yml)
     */
    public function __construct($baseUrl)
    {
        $this->baseUrl = $baseUrl;
        $client = new Client(['base_uri' => $this->baseUrl]);
        $this->client = $client;
    }

    /**
     * @When /^I request "([^"]*)"$/
     */
    public function iRequest($uri)
    {
        $request = $this->client->get($uri);
        $this->response = $request;
    }

    /**
     * @Then /^the response should be JSON$/
     */
    public function theResponseShouldBeJson()
    {
        $data = json_decode($this->response->getBody(true));

        if (empty($data)) {
            throw new \Exception("Response was not JSON\n" . $this->response);
       }
    }

    /**
     * @Then /^the response should be JSON equals "([^"]*)"$/
     */
    public function theResponseShouldBeJsonEquals($json)
    {
        $response = $this->response->getBody(true);

        if ($response != $json) {
            throw new \Exception('Response JSON was not equals to ' . $json);
        }
    }

    /**
     * @Then /^the response status code should be (\d+)$/
     */
    public function theResponseStatusCodeShouldBe($httpStatus)
    {
        if ((string)$this->response->getStatusCode() !== $httpStatus) {
            throw new \Exception('HTTP code does not match '.$httpStatus.
                ' (actual: '.$this->response->getStatusCode().')');
        }
    }

    /**
     * @Given /^the response has a "([^"]*)" property$/
     */
    public function theResponseHasAProperty($propertyName)
    {
        $data = json_decode($this->response->getBody(true));

        if (!empty($data)) {
            if (!isset($data->$propertyName)) {
                throw new \Exception("Property '".$propertyName."' is not set!\n");
            }
        } else {
            throw new \Exception("Response was not JSON\n" . $this->response->getBody(true));
        }
    }
     /**
     * @Then /^the "([^"]*)" property equals "([^"]*)"$/
     */
    public function thePropertyEquals($propertyName, $propertyValue)
    {
        $data = json_decode($this->response->getBody(true));

        if (!empty($data)) {
            if (!isset($data->$propertyName)) {
                throw new \Exception("Property '".$propertyName."' is not set!\n");
            }
            if ($data->$propertyName !== $propertyValue) {
                throw new \Exception('Property value mismatch! (given: '.$propertyValue.', match: '.$data->$propertyName.')');
            }
        } else {
            throw new \Exception("Response was not JSON\n" . $this->response->getBody(true));
        }
    }

    /**
     * @Then /^the response contains (\d+) item(s)$/
     */
    public function theResponseContainsItems($items)
    {
        $data = json_decode($this->response->getBody(true));

        if (!empty($data)) {
            if (count($data) != $items) {
                throw new \Exception("No contains $items items");
            }
        } else {
            throw new \Exception("Response was not JSON\n" . $this->response->getBody(true));
        }
    }
}


src/main/Hola/bootstrap.php
<?php
/**
 * <strong>Name : bootstrap.php</strong><br>
 * <strong>Desc : Bootstrapping application</strong><br>
 *
 * PHP version 5.3
 *
 * @category  Hello-canada
 * @package   Hello-canada
 * @author    Desarrollo <desarrollo@hola-internet.com>
 * @copyright 2013 Hola.com
 * @license   Apache 2 License http://www.apache.org/licenses/LICENSE-2.0.html
 * @version   GIT: $Id$
 * @link      http://www.hola.com
 * @since     File available since Release 0.1
 */

date_default_timezone_set('UTC');

putenv('PHP_ENV=testing');

$php_env = getenv('PHP_ENV');

if (empty($php_env)) {
    $php_env = 'testing';
    putenv("PHP_ENV={$php_env}");
}

// Checking composer autoload.
// Previous to ConfigMediator as the latter is spread across the former.
$loader = include __DIR__ . '/../../../vendor/autoload.php';

$loader->setUseIncludePath(true);

// Instantiating Config Mediator
$configuration = new Hola\Configuration\ConfigMediator(__DIR__ . "/../../../");

// Setting our common variables.
$configuration->pushConfigurationFromFiles(
    array(__DIR__ . "/../resources/config/config.yml"),
    array("common")
);

// Setting our environment variables.
$configuration->pushConfigurationFromFiles(
    array(
        $configuration->getProperty(
            "CONFIGURATION_FILES_".
            strtoupper($php_env)
        )
    ),
    array($php_env)
);

set_include_path(
    '.' .
    PATH_SEPARATOR . $configuration->getProperty("VENDOR_PATH") .
    PATH_SEPARATOR . $configuration->getProperty("MOCKS_PATH") .
    PATH_SEPARATOR . get_include_path()
);

$_SERVER['REQUEST_METHOD'] = 'GET';
